### PR TITLE
Move the dynamic trees size setting to the forester's hut

### PR DESCRIPTION
**Pull request by MotionlessTrain**
Closes a discussion on discord
Closes #
Closes #

# Changes proposed in this pull request:
- Create a new setting in the Forester's hut for the dynamic trees radius
- It uses the current config setting as default value and is only visible when dynamic trees is actually loaded
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobFindTree.java [hunk 1]
```diff
@@ -154,7 +158,14 @@ private boolean isNearTree(@NotNull final Node n)
 
     private boolean isTree(final BlockPos pos)
     {
-        if (Tree.checkTree(world, pos, excludedTrees) && Tree.checkIfInColonyAndNotInBuilding(pos, colony))
+        final LivingEntity entity = this.entity.get();
+        BuildingLumberjack building = null;
```
Comment (MotionlessTrain) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobFindTree.java]: I ended up getting the lumberjack building from the entity. That seemed to be the easiest solution

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java [hunk 2]
```diff
@@ -352,9 +353,13 @@ public static boolean checkTree(@NotNull final IWorldReader world, final BlockPo
         }
 
         // Only harvest nearly fully grown dynamic trees(8 max)
+        int treeSize = 0;
+        if(building != null) {
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java]: formatting. And why is the building nullable?

Comment (MotionlessTrain) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java]: To account for problems when the forester gets fired while trying to path to a tree (his worker hut probably isn’t a BuildingLumberjack anymore in that case)

Comment (MotionlessTrain) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java]: Looking through the code once again, I found a better way to get the setting over here

**Review by marchermans — APPROVED** [review]

**Pull request merged**